clients/node/client-rds-data-node/protocol/AwsRestJson1_1.ts

```diff
@@ -83,49 +88,36 @@ export function executeStatementAwsRestJson1_1Deserialize(
 function executeStatementAwsRestJson1_1DeserializeError(
   output: HttpResponse
 ): Promise<ExecuteStatementResponse> {
-  let data = parseBody(output);
-  if (output.statusCode === 400 && data.dbConnectionId !== undefined) {
-    return Promise.reject({
-      __type: "com.amazon.rdsdataservice#StatementTimeoutException",
-      $name: "StatementTimeoutException",
-      $fault: "client",
-      message: data.message,
-      dbConnectionId: data.dbConnectionId
-    });
-  }
-
-  if (output.statusCode === 400) {
-    return Promise.reject({
-      __type: "com.amazon.rdsdataservice#BadRequestException",
-      $name: "BadRequestException",
-      $fault: "client",
-      message: data.message
-    });
-  }
-
-  if (output.statusCode === 403) {
-    return Promise.reject({
-      __type: "com.amazon.rdsdataservice#ForbiddenException",
-      $name: "ForbiddenException",
-      $fault: "client",
-      message: data.message
-    });
-  }
-
-  if (output.statusCode === 500) {
-    return Promise.reject({
-      __type: "com.amazon.rdsdataservice#InternalServerErrorException",
-      $name: "InternalServerErrorException",
-      $fault: "server"
-    });
-  }
+  const data = parseBody(output);
+
+  const errorMap: { [index: string]: any } = {
+    BadRequestException: badRequestExceptionDeserialize(data),
+    StatementTimeoutException: statementTimeoutExceptionDeserialize(data),
+    ForbiddenException: forbiddenExceptionDeserialize(data),
+    InternalServerErrorException: internalServerErrorExceptionDeserialize(data),
+    ServiceUnavailableError: serviceUnavailableErrorDeserialize(data),
+    "com.amazon.rdsdataservice#BadRequestException": badRequestExceptionDeserialize(
+      data
+    ),
+    "com.amazon.rdsdataservice#StatementTimeoutExceptio": statementTimeoutExceptionDeserialize(
+      data
+    ),
+    "com.amazon.rdsdataservice#ForbiddenException": forbiddenExceptionDeserialize(
+      data
+    ),
+    "com.amazon.rdsdataservice#InternalServerErrorException": internalServerErrorExceptionDeserialize(
+      data
+    ),
+    "com.amazon.rdsdataservice#ServiceUnavailableError": serviceUnavailableErrorDeserialize(
+      data
+    )
+  };
 
-  if (output.statusCode === 503) {
-    return Promise.reject({
-      __type: "com.amazon.rdsdataservice#ServiceUnavailableError",
-      $name: "ServiceUnavailableError",
-      $fault: "server"
-    });
+  if ("x-amzn-ErrorType" in output.headers) {
```

Comment:
Is this a case-insensitive lookup?

Reply:
it is. we need an abstraction to make that easier from the HttpResponse.